x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Subject.java

```diff
@@ -283,7 +283,7 @@ RoleReference.ApiKeyRoleReference buildRoleReferenceForCrossClusterApiKey() {
         final BytesReference limitedByRoleDescriptorsBytes = (BytesReference) metadata.get(API_KEY_LIMITED_ROLE_DESCRIPTORS_KEY);
         assert isEmptyRoleDescriptorsBytes(limitedByRoleDescriptorsBytes)
             : "cross cluster API keys must have empty limited-by role descriptors";
-        return new RoleReference.ApiKeyRoleReference(apiKeyId, roleDescriptorsBytes, RoleReference.ApiKeyRoleType.ASSIGNED);
+        return new RoleReference.ApiKeyRoleReference(apiKeyId, roleDescriptorsBytes, RoleReference.ApiKeyRoleType.ASSIGNED, true);
```

Comment:
The boolean flag is ugly, but it allows for a much smaller change. I'd like to refactor this in the (near) future but for now this approach allows us to get validation in the right place without introducing larger code-changes.